graphics/opensubdiv/Portfile

```diff
@@ -0,0 +1,87 @@
+license             Apache-2
+maintainers         @jasonliu-- openmaintainer
+homepage            http://graphics.pixar.com/opensubdiv
+description         subdivision surface library
+long_description    OpenSubdiv is a set of libraries developed by \
+                    Pixar Animation Studios that implement \
+                    high-performance subdivision surface (subdiv) \
+                    evaluation on massively-parallel CPU and GPU \
+                    architectures. This code path is optimized for \
+                    drawing deforming surfaces with static topology at \
+                    interactive framerates.
+
+checksums           rmd160  1ce640b559ffac0e0bbfd817e77b9f7991bd326d \
+                    sha256  6b09b774b15cb57ac9b1f3c8378a8dbd077eb0ee349d0e8d2cb12ffd52777084 \
+                    size    39232618
+
+depends_build-append \
+                    port:doxygen \
+                    port:py-docutils \
+                    port:py-pygments \
+                    port:graphviz
+depends_lib         port:tbb \
+                    port:glfw \
+                    port:ptex \
+                    port:zlib
+
+post-patch {
+    # OpenSubdiv creates a framework bundle,
+    # but puts it in the wrong place.
+    reinplace "s|@rpath|${frameworks_dir}|g" \
+        ${worksrcpath}/opensubdiv/CMakeLists.txt
+    reinplace -E "s|\(FRAMEWORK DESTINATION\).*|\\1 \"${frameworks_dir}\"|g" \
+        ${worksrcpath}/opensubdiv/CMakeLists.txt
+
+    # Make it so that OpenSubdiv is able to find MacPorts' Docutils.
+    if {![file exists ${prefix}/bin/rst2html.py]} {
+        set files [glob -directory ${prefix}/bin "rst2html-*.py"]
+        foreach f $files {
+            if {[regexp {rst2html-.*\.py} $f result]} {
+                reinplace "s|rst2html.py|${result}|g" \
+                    ${worksrcpath}/cmake/FindDocutils.cmake
+            }
+        }
+    }
+    # Set a project name, otherwise Doxygen will generate
+    # documentation with a project name of "My Project".
+    reinplace -E "s|^\(PROJECT_NAME.*\)$|\\1 OpenSubdiv|" \
+        ${worksrcpath}/documentation/OpenSubdiv.doxy
+}
+
+# By default, OpenSubdiv puts all examples, tutorials, regression
+# tests, etc. directly into the ${prefix}/bin directory. We instead
+# place them into a location that makes a bit more sense.
+#                           vvvvvv
+configure.args      -DCMAKE_BINDIR_BASE=${prefix}/share/${name} \
+                    -DNO_CUDA=1 \
+                    -DNO_CLEW=1
+if {${os.major} < 16} {
+    # OpenSubdiv uses features of Metal that were added starting
+    # in macOS 10.12 (e.g. MTLTessellationPartitionMode)
+    configure.args-append   -DNO_METAL=1
+}
+
+variant clew description {Enable CLEW wrapper library} {
+    depends_lib-append      port:clew
+    configure.args-replace  -DNO_CLEW=1 -DNO_CLEW=0
+}
+
+variant glew description {Enable use of GLEW} {
+    depends_lib-append      port:glew
+    configure.args-append   -DNO_GLEW=0
```

Comment:
would it make sense to be default pass `-DNO_GLEW=1` in the configure args., and then here in this variant replace it with `=0`. i.e. as per CLEW support above.

Reply:
`-DNO_GLEW=1` is the default value set by the source code, so it isn't needed. The CLEW support is set to `-DNO_CLEW=0` by default, so I have to override it. Don't ask me why the devs do it this way... in fact, it seems that the devs changed these default settings between versions (v3.2.0 vs. v3.4.3).

Reply:
even if its the default its better from our side if you explicitly set it, as then we are safe to upstream changing their minds in the future. So please do this.

Reply:
I would still prefer if you explicitly passed `-DNO_GLEW=1` and didn't reply on the default settings. Your comment above "it seems that the devs changed these default settings between versions (v3.2.0 vs. v3.4.3)" shows why as then we are not reliant on what they choose.

Reply:
I'm going to resolve this as otherwise the port looks OK, and this isn't enough to hold it up getting merged. Please consider the above though for a future update.